images/sds-replicated-volume-controller/src/pkg/controller/linstor_resources_watcher.go

```diff
@@ -442,6 +552,13 @@ func GetListPV(ctx context.Context, cl client.Client) ([]core.PersistentVolume,
 	if err != nil {
 		return nil, err
 	}
+
+	for i := range PersistentVolumeList.Items {
+		pv := &PersistentVolumeList.Items[i]
+		if pv.Labels == nil {
+			pv.Labels = make(map[string]string)
+		}
+	}
 	return PersistentVolumeList.Items, nil
 }
```

Comment:
`UpdatePV` really does nothing, you can remove it and just use `cl.Update(ctx, pv)` everywhere